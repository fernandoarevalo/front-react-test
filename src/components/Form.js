import React from "react";
import { useForm } from "react-hook-form";
import M from "materialize-css";
import Modal from "components/Modal";
// Utils
import { FORM } from "utils/constants/Contants";

export default function Form() {
  const { register, handleSubmit, errors, reset } = useForm();

  const onSubmit = (data) => {
    console.log(data);
    const elems = document.getElementById("modal");
    const instance = M.Modal.init(elems, {});
    instance.open();
    setTimeout(() => {
      instance.close();
      reset();
    }, 5000);
  };

  return (
    <React.Fragment key="form">
      <section className="container">
        <div className="row">
          <div className="col s12 l6 offset-l3">
            <form onSubmit={handleSubmit(onSubmit)} className="card-panel">
              <div className="input-field">
                <input
                  id="name"
                  name="name"
                  type="text"
                  className="validate"
                  ref={register({
                    required: {
                      value: true,
                      message: FORM.REQUIRED,
                    },
                  })}
                />
                <label htmlFor="last_name">{FORM.NAME}</label>
                {errors.name && (
                  <span className="red-text">{errors.name.message}</span>
                )}
              </div>
              <div className="input-field">
                <input
                  id="email"
                  name="email"
                  type="email"
                  className="validate"
                  ref={register({
                    required: {
                      value: true,
                      message: FORM.REQUIRED,
                    },
                  })}
                />
                <label htmlFor="email">{FORM.EMAIL}</label>
                {errors.email && (
                  <span className="red-text">{errors.email.message}</span>
                )}
              </div>
              <div className="input-field">
                <input
                  id="cellphone"
                  name="cellphone"
                  type="text"
                  className="validate"
                  ref={register({
                    required: {
                      value: true,
                      message: FORM.REQUIRED,
                    },
                  })}
                />
                <label htmlFor="cellphone">{FORM.CELLPHONE}</label>
                {errors.cellphone && (
                  <span className="red-text">{errors.cellphone.message}</span>
                )}
              </div>
              <div className="input-field">
                <input
                  id="age"
                  name="age"
                  type="number"
                  className="validate"
                  ref={register({
                    required: {
                      value: true,
                      message: FORM.REQUIRED,
                    },
                  })}
                />
                <label htmlFor="age">{FORM.AGE}</label>
                {errors.age && (
                  <span className="red-text">{errors.age.message}</span>
                )}
              </div>
              <button type="submit" className="waves-effect waves-light btn">
                Enviar
              </button>
            </form>
          </div>
        </div>
      </section>

      <div id="modal" className="modal">
        <Modal />
      </div>
    </React.Fragment>
  );
}
