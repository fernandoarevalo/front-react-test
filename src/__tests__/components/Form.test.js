import React from "react";
import { render, cleanup, fireEvent, screen } from "@testing-library/react";
import Form from "components/Form";

describe("<Form />", () => {
  afterEach(() => {
    cleanup();
  });

  test("Snapshot Form component", () => {
    const { asFragment } = render(<Form />);
    expect(asFragment()).toMatchSnapshot();
  });
});
