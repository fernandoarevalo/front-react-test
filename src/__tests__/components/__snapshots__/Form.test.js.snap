// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> Snapshot Form component 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col s12 l6 offset-l3"
      >
        <form
          class="card-panel"
        >
          <div
            class="input-field"
          >
            <input
              class="validate"
              id="name"
              name="name"
              type="text"
            />
            <label
              for="last_name"
            >
              Nombre
            </label>
          </div>
          <div
            class="input-field"
          >
            <input
              class="validate"
              id="email"
              name="email"
              type="email"
            />
            <label
              for="email"
            >
              Email
            </label>
          </div>
          <div
            class="input-field"
          >
            <input
              class="validate"
              id="cellphone"
              name="cellphone"
              type="text"
            />
            <label
              for="cellphone"
            >
              Celular
            </label>
          </div>
          <div
            class="input-field"
          >
            <input
              class="validate"
              id="age"
              name="age"
              type="number"
            />
            <label
              for="age"
            >
              Edad
            </label>
          </div>
          <button
            class="waves-effect waves-light btn"
            type="submit"
          >
            Enviar
          </button>
        </form>
      </div>
    </div>
  </section>
  <div
    class="modal"
    id="modal"
  >
    <div>
      <div
        class="modal-content"
      >
        <h4>
          Grandioso!
        </h4>
        <p>
          Tu información fue enviada con éxito, estaremos en contacto contigo
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
